### PR TITLE
Add account settings page with ability to add email to account

### PR DESCRIPTION
**Pull request by wraeth-eth**
## What does this PR do and why?

Front end changes to integrate database authorization and email settings

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by wraeth-eth** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3156** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3156" target="_blank">Graphite</a> 
    * **PR #3157** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3157" target="_blank">Graphite</a> 
      * **PR #3158** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3158" target="_blank">Graphite</a> 
        * **PR #3161** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3161" target="_blank">Graphite</a> 
          * **PR #3173** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3173" target="_blank">Graphite</a> 
            * **PR #3174** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3174" target="_blank">Graphite</a>  👈
        * **PR #3159** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3159" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3174?utm_source=stack-comment).

File: src/components/AccountSettingsDashboard/AccountSettingsDashboard.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      values: AccountSettingsFormType,
+      helpers: FormikHelpers<AccountSettingsFormType>,
+    ) => {
+      try {
+        if (!values.email) throw new Error('no email')
+        await axios.post('/api/account/add-email', {
+          email: values.email,
+        })
+        emitInfoNotification(
+          `Request sent! Please check ${values.email} and verify your new email address.`,
```
Comment (tomquirk) [inline src/components/AccountSettingsDashboard/AccountSettingsDashboard.tsx:39]:
```suggestion
          `Check ${values.email} and verify your new email address.`,
```

File: src/components/AccountSettingsDashboard/AccountSettingsDashboard.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+  const onEmailResendClicked = useCallback(async () => {
+    if (!lastEmailUpdated)
+      emitErrorNotification('Something has gone wrong, please contact support.')
+
+    await axios.post('/api/account/add-email', {
+      email: lastEmailUpdated,
+    })
+    emitInfoNotification(
+      `Request sent! Please check ${lastEmailUpdated} and verify your new email address.`,
```
Comment (tomquirk) [inline src/components/AccountSettingsDashboard/AccountSettingsDashboard.tsx:62]:
```suggestion
      `Check ${lastEmailUpdated} and verify your new email address.`,
```

File: src/pages/account/[addressOrEnsName]/settings.page.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      redirect: {
+        destination: '/',
+        permanent: false,
+      },
+    }
+  }
+
+  const user = await supabase
+    .from('users')
+    .select('*')
```
Comment (tomquirk) [inline src/pages/account/[addressOrEnsName]/settings.page.tsx:34]: same mebe

**Comment by wraeth-eth** [comment]
@wraeth-eth queued this pull request to merge with [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3174).

**Comment by wraeth-eth** [comment]
@wraeth-eth merged this pull request with [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3174).

**Pull request merged**